src/script/util/ComponentUtil.ts

```diff
@@ -20,30 +20,30 @@
 import ko from 'knockout';
 import {container, InjectionToken} from 'tsyringe';
 
-export function registerReactComponent(
+export function registerReactComponent<Props>(
   name: string,
   {
     template,
     component,
     optionalParams = [],
     injected = {},
   }: {
-    component: React.ComponentType;
+    component: React.ComponentType<Props>;
     injected?: Record<string, InjectionToken>;
-    optionalParams?: string[];
+    optionalParams?: (keyof Props)[];
     template: string;
   },
 ) {
   ko.components.register(name, {
-    template: template,
-    viewModel: function (knockoutParams: Record<string, any>) {
+    template,
+    viewModel: function (knockoutParams: Record<keyof Props, any>) {
```

Comment:
Would that also work?
```suggestion
    viewModel: function (knockoutParams: Props) {
```

Reply:
Good idea!